test/node/parser/deserializer.test.ts

```diff
@@ -0,0 +1,13 @@
+import * as BSON from '../../register-bson';
+import { expect } from 'chai';
+
+describe('deserializer()', () => {
+  it('should only enumerate own property keys from input options', () => {
+    const bytes = BSON.serialize({ someKey: [1] });
+    const options = { fieldsAsRaw: { someKey: true } };
+    Object.setPrototypeOf(options, { promoteValues: false });
+    const result = BSON.deserialize(bytes, options);
+    expect(result).to.have.property('someKey').that.is.an('array');
+    expect(result.someKey[0]).to.not.have.property('_bsontype', 'Int32');
```

Comment:
related to the comment above - this test kind of confusing.  It's not immediately apparent why `fieldsAsRaw` is necessary when looking at the test.  I had to read the code to realize that we only use the for-in loop when `fieldsAsRaw` is set to true, so we need that option to trigger the offending code.

```typescript
context('when the fieldsAsRaw options is present and has a value that corresponds to a key in the object', () => {
  it('ignores non-own properties set on the options object', () => {
	const bytes = BSON.serialize({ someKey: [1] });
    const options = { fieldsAsRaw: { someKey: true } };
    Object.setPrototypeOf(options, { promoteValues: false });
    const result = BSON.deserialize(bytes, options);
    expect(result).to.have.property('someKey').that.is.an('array');
    expect(result.someKey[0], 'expected promoteValues option set on options object prototype to be ignored, but it was not').to.not.be.instanceOf(Int32);
  });
});
```